netmanager-app/core/redux/Airqlouds/operations.ts

```diff
@@ -0,0 +1,182 @@
+          show: true
+        })
+      );
+    });
+};
+
+export const setCurrentAirQloudData = (airqloud) => (dispatch, getState) => {
+  const tenant = getState().organisation.name;
+  const newAirqloud = createAirqloudSiteOptions(airqloud);
+  const currentAirqloudState = JSON.parse(localStorage[CURRENT_AIRQLOUD_KEY] || '{}');
+  localStorage.setItem(
+    CURRENT_AIRQLOUD_KEY,
+    JSON.stringify({ ...currentAirqloudState, [tenant]: newAirqloud })
+  );
+  dispatch({
+    type: SET_CURRENT_AIRQLOUD_SUCCESS,
+    payload: createAirqloudSiteOptions(newAirqloud)
+  });
+};
+
+export const setDefaultAirQloud = () => async (dispatch, getState) => {
+  const tenant = getState().organisation.name;
+  const airqloud = JSON.parse(localStorage[CURRENT_AIRQLOUD_KEY] || '{}')[tenant];
+  if (airqloud) {
+    dispatch({
+      type: SET_CURRENT_AIRQLOUD_SUCCESS,
+      payload: airqloud
+    });
+  } else {
+    const response = await getAirQloudsApi({});
+    if (!response || isEmpty(response.airqlouds)) return;
+    const { airqlouds } = response;
+    const current = airqlouds.filter((airqloud) => airqloud.long_name === 'Uganda');
+    dispatch(setCurrentAirQloudData((current.length > 0 && current[0]) || airqlouds[0]));
+    dispatch({
+      type: LOAD_ALL_AIRQLOUDS_SUCCESS,
+      payload: transformArray(airqlouds, '_id')
+    });
+  }
+};
+
+export const fetchDashboardAirQloudsData = (options) => async (dispatch) => {
+  return await getDashboardAirQloudsApi()
+    .then((resData) => {
+      if (isEmpty(resData.airqlouds || [])) return;
+      dispatch({
+        type: LOAD_DASHBOARD_AIRQLOUDS_SUCCESS,
+        payload: transformArray(resData.airqlouds, '_id')
+      });
+      if (options && options.callable instanceof Function) options.callable();
+    })
+    .catch((err) => {
+      dispatch({
+        type: LOAD_DASHBOARD_AIRQLOUDS_FAILURE,
+        payload: err
+      });
+      if (options && options.onerror instanceof Function) options.onerror();
+    });
+};
+
+export const getAirqloudDetails = (airQloudID) => async (dispatch) => {
+  return await getAirQloudsApi({ id: airQloudID })
+    .then((resData) => {
+      dispatch({
+        type: LOAD_SELECTED_AIRQLOUD_SUCCESS,
+        payload: resData.airqlouds[0]
+      });
+    })
+    .catch((err) => {
+      dispatch({
+        type: LOAD_SELECTED_AIRQLOUD_FAILURE,
+        payload: err
+      });
+    });
+};
+
+export const removeAirQloudData = () => (dispatch) => {
+  dispatch({
+    type: REMOVE_SELECTED_AIRQLOUD_SUCCESS,
+    payload: {}
+  });
+};
+
+export const fetchAirqloudsSummaryData = () => async (dispatch) => {
+  return await getAirqloudsSummaryApi()
+    .then((resData) => {
+      dispatch({
+        type: LOAD_AIRQLOUDS_SUMMARY_SUCCESS,
+        payload: resData.airqlouds
+      });
+    })
+    .catch((err) => {
+      dispatch({
+        type: LOAD_AIRQLOUDS_SUMMARY_FAILURE,
+        payload: err
+      });
+    });
+};
```

Comment:
Please avoid copying and pasting things from netmanager to the new nextjs project as they are. Follow the current convention in this nextjs app that I configured of utilising slices, api calls should be done in hooks and then we can also update the redux states within those hooks. Don't just copy and paste please, and only add the code that you need. Otherwise we will have so much unused code yet this is a new project. @danielmarv